pallets/nfts/src/features/transfer.rs

```diff
@@ -87,6 +87,24 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		// Perform the transfer with custom details using the provided closure.
 		with_details(&collection_details, &mut details)?;
 
+		// Update account balance of the owner.
+		let owner_balance =
+			AccountBalance::<T, I>::mutate(collection, &details.owner, |balance| -> u32 {
+				balance.saturating_dec();
+				*balance
+			});
+		if owner_balance == 0 {
+			collection_details.item_holders.saturating_dec();
+		}
+		// Update account balance of the destination account.
+		let dest_balance = AccountBalance::<T, I>::mutate(collection, &dest, |balance| -> u32 {
+			balance.saturating_inc();
+			*balance
+		});
+		if dest_balance == 1 {
+			collection_details.item_holders.saturating_inc();
+		}
+
```

Comment:
Not following the second check for both owner_balance and dest_balance?

Reply:
Suppose `owner_balance` is the balance of ALICE and `dest_balance` is the balance of BOB after transferring. There are three cases: 
- `ALICE = 0` and `BOB = 1` => ALICE is no longer the collection item holder and BOB is the new collection holder.
- `ALICE = 0` and `BOB > 1` => ALICE is no longer the collection item holder and BOB is the existing collection holder. Decrease the `item_holders` counter.
- `ALICE > 0` and `BOB = 1` => ALICE is still the collection item holder and BOB is the new collection holder. Increase the `item_holders` counter.

Reply:
Outdated